src/dev.ts

```diff
@@ -0,0 +1,279 @@
+
+interface DevOptions {
+  port?: number
+  dashboardPort?: number
+  'watch-contracts'?: boolean
+  'hot-reload'?: boolean
+  'auto-test'?: boolean
+  'interactive'?: boolean
+  debug?: boolean
+}
+
+interface ContractChangeEvent {
+  file: string
+  type: 'added' | 'changed' | 'removed'
+  timestamp: Date
+}
+
+interface DevMetrics {
+  contractsWatched: number
+  changesDetected: number
+  hotReloads: number
+  uptime: number
+}
+
+/**
+ * Live Development Environment for Chelonia Contracts
+ *
+ * Provides live-testing functionality similar to grunt dev:
+ * - Hot reload of contracts when files change
+ * - Interactive testing environment with running server
+ * - Automatic contract redeployment on changes
+ * - Live dashboard for monitoring contract changes
+ *
+ * This leverages the robust chel serve infrastructure for reliability.
+ */
+class DevEnvironment {
+  private watchers: { close?: () => void }[] = []
+  private serverProcess?: unknown
+  private options: DevOptions
+  private projectRoot: string
+  private metrics: DevMetrics
+  private startTime: Date
+  private contractChanges: ContractChangeEvent[] = []
+
+  constructor (projectRoot: string, options: DevOptions) {
+    this.projectRoot = resolve(projectRoot)
+    this.options = options
+    this.metrics = {
+      contractsWatched: 0,
+      changesDetected: 0,
+      hotReloads: 0,
+      uptime: 0
+    }
+    this.startTime = new Date()
+  }
+
+  // Simple fast non-crypto hash for content (djb2)
+  private hashContent (str: string): string {
+    let h = 5381
+    for (let i = 0; i < str.length; i++) h = ((h << 5) + h) ^ str.charCodeAt(i)
+    // force to unsigned 32-bit and to hex
+    return (h >>> 0).toString(16)
+  }
+
+  async start () {
+    this.printWelcomeBanner()
+    await this.startServer()
+    await this.startContractWatching()
+    await this.startInteractiveMode()
+  }
+
+  private async startServer () {
+    console.log(colors.blue('🚀 Starting development server with hot reload...'))
+
+    // Start the server in the background using our robust serve infrastructure
+    const serverArgs = [
+      '--dp', String(this.options.dashboardPort || 3000),
+      '--port', String(this.options.port || 8000),
+      '--db-type', 'mem', // Use memory for faster dev iterations
+      this.projectRoot
+    ]
+
+    try {
+      // Start the serve function in the background
+      // Don't await it here as it runs indefinitely
+      this.serverProcess = serve(serverArgs).catch(error => {
+        console.error(colors.red('❌ Server error:'), error)
+        process.exit(1)
+      })
+
+      // Give the server a moment to start
+      await new Promise(resolve => setTimeout(resolve, 2000))
+
+      console.log(colors.green('✅ Development server started'))
+      console.log(colors.gray(`   Dashboard: http://localhost:${this.options.dashboardPort || 3000}`))
+      console.log(colors.gray(`   Application: http://localhost:${this.options.port || 8000}`))
```

Comment:
These lines made me realise two things:

1. For the dashboard, this is accurate because you have `host: 'localhost'` in `dashboard-server.ts`. However, likely you shouldn't have that line, or have it be configurable.
2. For the application, this is inaccurate because no `host` is given to `Hapi`, meaning that it'll listen on all addresses.